### PR TITLE
[BugFix] Exit if create thread failed when submit apply task

### PR DESCRIPTION
**Pull request by decster**
Fixes #issue

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5
  - [x] 2.4

### CODE REVIEW

**Review by trueeyu — APPROVED** [review]

File: be/src/storage/tablet_updates.cpp [hunk 1]
```diff
@@ -745,8 +745,7 @@ void TabletUpdates::_check_for_apply() {
     if (!st.ok()) {
         std::string msg =
                 strings::Substitute("submit apply task failed: $0 $1", st.to_string(), _debug_string(false, false));
-        LOG(ERROR) << msg;
-        _set_error(msg);
+        LOG(FATAL) << msg;
```
Comment (trueeyu) [inline be/src/storage/tablet_updates.cpp:748]: LOG(FATAL) if thread queue is full ?

**Review by sevev — APPROVED** [review]

File: be/src/storage/tablet_updates.cpp (hunk 1 above)
Comment (decster) [reply be/src/storage/tablet_updates.cpp:748]: yes

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Pull request merged**